src/replica/HttpAsyncClientApp.cc

```diff
@@ -0,0 +1,204 @@
+
+// System headers
+#include <algorithm>
+#include <fstream>
+#include <iostream>
+#include <set>
+#include <stdexcept>
+#include <vector>
+
+// Third-party headers
+#include "boost/asio.hpp"
+#include "nlohmann/json.hpp"
+
+// Qserv headers
+#include "replica/HttpAsyncClient.h"
+
+using namespace std;
+using json = nlohmann::json;
+
+namespace {
+
+string const description =
+    "This application sends requests to a Web server over the HTTP/HTTPS protocol"
+    " using the asynchronous client API. If option '--file=<file>' is present the result"
+    " will be writted to the specified file. Otherwise the content will be printed to"
+    " the standard output stream.";
+
+bool const injectDatabaseOptions = false;
+bool const boostProtobufVersionCheck = false;
+bool const enableServiceProvider = false;
+
+} /// namespace
+
+namespace lsst {
+namespace qserv {
+namespace replica {
+
+HttpAsyncClientApp::Ptr HttpAsyncClientApp::create(int argc, char* argv[]) {
+    return Ptr(new HttpAsyncClientApp(argc, argv));
+}
+
+
+HttpAsyncClientApp::HttpAsyncClientApp(int argc, char* argv[])
+    :   Application(
+            argc, argv,
+            ::description,
+            ::injectDatabaseOptions,
+            ::boostProtobufVersionCheck,
+            ::enableServiceProvider
+        ) {
+
+    parser().required(
+        "url",
+        "The URL to read data from.",
+        _url
+    ).option(
+        "method",
+        "The HTTP method. Allowed values: GET, POST, PUT, DELETE.",
+        _method
+    ).option(
+        "header",
+        "The HTTP header to be sent with a request. Note this test application allows"
+        " only one header. The format of the header is '<key>[:<val>]'.",
+        _header
+    ).option(
+        "data",
+        "The data to be sent in the body of a request.",
+        _data
+    ).option(
+        "max-response-data-size",
+        "The maximum size (bytes) of the response body. If a value of the parameter is set"
+        " to 0 then the default limit of 8M imposed by the Boost.Beast library will be assumed.",
+        _maxResponseBodySize
+    ).option(
+        "expiration-ival-sec",
+        "A timeout to wait before the completion of a request. The expiration timeout includes"
+        " all phases of the request's execution, including establishing a connection"
+        " to the server, sending the request and waiting for the server's response."
+        " If a value of the parameter is set to 0 then no expiration timeout will be"
+        " assumed for the request.",
+        _expirationIvalSec
+    ).option(
+        "file",
+        "A path to an output file where the response body received from a remote source will"
+        "  be written. This option is ignored if the flag --body is not specified.",
+        _file
+    ).flag(
+        "result2json",
+        "If specified the flag will cause the application to interpret the response body as"
+        " a JSON object.",
+        _result2json
```

Comment:
Is this actually used?

Reply:
It was _meant_ to be used. I missed implementing a small piece of code that would do it. Thank you for noticing!